### PR TITLE
Add missing Dataset field serializers

### LINKED ISSUE #377: Add field_serializers for Dataset field without serializable types
https://github.com/NOAA-OWP/DMOD/blob/12ccd5ea4ccffc51615c621657b127b509f54fb5/python/lib/core/dmod/core/dataset.py#L68-L80

The following fields need `field_serializers` methods:

- `manager_uuid`
- `expires`
- `created_on`
- `last_updated`

### PR DESCRIPTION
**Pull request by aaraney**
fixes #377

## Changes

- Add missing `dmod.core.dataset.Dataset` field serializers (fixes #377).

### CODE REVIEW

**Comment by aaraney** [comment]
This depends on #381

**Comment by robertbartel** [comment]
@aaraney, I've marked the issue as no longer blocked.  Probably this just needs the branch rebased.

**Comment by aaraney** [comment]
Thanks for the review, @robertbartel!

**Comment by aaraney** [comment]
@robertbartel, rebased and ready for review! Thanks in advance!

File: python/lib/core/dmod/core/dataset.py [hunk 1]
```diff
@@ -117,7 +117,19 @@ class Config:
...
         arbitrary_types_allowed = True
-        field_serializers = {"uuid": lambda f: str(f)}
+
+        def _serialize_datetime(self: "Dataset", value: Optional[datetime]) -> Optional[str]:
+            if isinstance(value, datetime):
+                return value.strftime(self._SERIAL_DATETIME_STR_FORMAT)
+            return value
+
+        field_serializers = {
+            "uuid": str,
```
Comment (robertbartel) [inline python/lib/core/dmod/core/dataset.py:127]: I think `uuid` should have something akin to the lambda used for `manager_uuid`, although perhaps it's declaration needs adjustment.  That is:  

``` Python
uuid: Optional[UUID] = Field(default_factory=uuid4)
```

If I follow correctly, this implies the possibility of explicitly receiving ``None``.  It's possible there's a subtlety about Pydantic I'm missing that _requires_ this to be declared as _Optional_ even though it really isn't (although if that's the case, it's bothersome).

**Review by robertbartel — COMMENTED** [review]
There's one thing that needs to be cleared up before I approve, although I probably am just missing something about how Pydantic works.

**Review by robertbartel — APPROVED** [review]

**Pull request merged**